Make interactive Dashboard 🕺

emilom opened the pull request:
Part of #30 

Made dashboard with react-grid-layout🍝

blauks committed:
Install evergreen

blauks committed:
Merge with master

blauks committed:
Add ThresholdChart as child to dashboard item

blauks committed:
Make small dashboard item pretty

blauks committed:
Merge with master

blauks committed:
Fix yarn.lock

blauks committed:
Finish small dashboard

blauks committed:
Make small dashboard item and chart dynamic

Commit:
first draft dashboard

Commit:
merge

Commit:
lagd et dashboard som delvis fungerer

Commit:
Make placeholder for dashboard

Commit:
Delete unnecessary files

Commit:
Make dashboard with types

Commit:
Add story for dashboard

Commit:
Fix lint errors

Commit:
fix sider errors

Commit:
fix sider errors, forgot comment

ghost commented:
✅ Errors or warnings were resolved via the commit 3f8d90a! Please visit [Sider](https://sider.review/gh/repos/336560183/pulls/46) for details.

---

You can [turn off this notification](https://sider.review/gh/repos/336560183/settings#notifications) if you don't want to receive it from now on.

emilom committed:
Update sider.yml

FredrikAugust commented:
Note that this PR appears to depend on #45.
I believe we should aim to resolve the issues in that PR, rebase that onto master, and then into this branch again before merging this PR.
I will review this PR shortly, but will not comment on the points discussed in #45, so refer to that for additional comments.

FredrikAugust commented on sider.yml:
```diff
@@ -23,7 +23,7 @@ linter:
         #       - bar.rb
         locale: UK
-#     ignore: center,behavior
+        ignore: center,behavior,color
```
Generally, try to avoid adding extra exceptions that aren't needed. Ref. "behavior".

FredrikAugust commented on src/components/molecules/DashboardItem.tsx:
```diff
@@ -0,0 +1,9 @@
+import { Card } from 'evergreen-ui';
+
+const DashboardItem: React.FC = () => {
+    console.log('title');
```
Don't use `console.log` in React components

FredrikAugust commented on src/components/molecules/DashboardItem.tsx:
```diff
@@ -0,0 +1,9 @@
+
+const DashboardItem: React.FC = () => {
+    console.log('title');
+    return <Card width={'100%'} height={'100%'} border="default" elevation={1}></Card>;
```
Don't need to evaluate the strings as they are _simply strings_.
```suggestion
    return <Card width="100%" height="100%" border="default" elevation={1} />;
```

This is because statements wrapped in `{}` are evaluated as JS expressions.

FredrikAugust commented on src/components/molecules/DashboardItemSmall.tsx:
```diff
@@ -0,0 +1,25 @@
+import { DashboardItemSmallType } from '../../dataTypes/DashboardItems/DashboardItemSmallType';
+import { Card, Pane, Heading } from 'evergreen-ui';
+
+const DashboardItemSmall: React.FC<DashboardItemSmallType> = ({ title, children, width, height, titleSize }) => {
```
See #45.

FredrikAugust commented on src/components/organisms/Dashboard.tsx:
```diff
@@ -0,0 +1,148 @@
+import DashboardItem from '../molecules/DashboardItem';
+import { DashboardItemSize, DashboardItemInfo, DashboardProps } from '../../dataTypes/Dashboard/DashboardTypes';
+
+import 'react-grid-layout/css/styles.css';
+import 'react-resizable/css/styles.css';
```
Include this in `index.tsx` to ensure they're only loaded once.

FredrikAugust commented on src/components/organisms/Dashboard.tsx:
```diff
@@ -0,0 +1,148 @@
+
+    return (
+        <Pane height={'100%'} width={'80%'} marginLeft={'10%'}>
+            {console.log('layout change')}
```
```suggestion
```

FredrikAugust commented on src/components/organisms/Dashboard.tsx:
```diff
@@ -0,0 +1,148 @@
+    }, [items]);
+
+    return (
+        <Pane height={'100%'} width={'80%'} marginLeft={'10%'}>
```
Prev comment.

FredrikAugust commented on src/components/organisms/Dashboard.tsx:
```diff
@@ -0,0 +1,148 @@
+};
+
+const Dashboard: React.FC<DashboardProps> = ({ items }) => {
+    const [layout, setLayout] = useState<Layout[]>();
```
```suggestion
    const [layout, setLayout] = useState<Layout[]>([]);
```
Default to empty layout instead of `undefined`/`null`.

FredrikAugust commented on src/components/organisms/Dashboard.tsx:
```diff
@@ -0,0 +1,148 @@
+                        useCSSTransforms={true}
+                        isResizable={false}
+                    >
+                        {layout &&
```
Remove conditional rendering with default `[]` in `setState`. No longer necessary.

FredrikAugust commented on src/dataTypes/Dashboard/DashboardTypes.ts:
```diff
@@ -0,0 +1,14 @@
+export enum DashboardItemSize {
+    small,
+    medium,
+    large,
```
```suggestion
    SMALL = 2,
    MEDIUM = 3,
    LARGE = 4,
```

Default is all-caps for enums.
Set the values for the enum values here instead of retrieving them later.

FredrikAugust commented on src/dataTypes/Dashboard/DashboardTypes.ts:
```diff
@@ -0,0 +1,14 @@
+    large,
+}
+
+export type DashboardProps = {
+    items: DashboardItemInfo[];
+};
```
Props should be in the same file as the component.

FredrikAugust commented on src/components/organisms/Dashboard.tsx:
```diff
@@ -0,0 +1,148 @@
+
+import 'react-grid-layout/css/styles.css';
+import 'react-resizable/css/styles.css';
+
+type Layout = {
+    i: string;
```
It seems like there are more conversions from int to string than reverse, so you could type this as an int.

And with that replace `Layout` with `Layout = Record<'i' | 'x' | 'y' | 'w' | 'h', number>;`.

FredrikAugust committed:
Set up storybook controls

Preferred way to handle inputs to stories

FredrikAugust commented on src/index.css:
```diff
@@ -0,0 +1,9 @@
+}
+
+body {
+    margin: 0;
```
Consider a CSS reset

FredrikAugust commented on src/index.css:
```diff
@@ -0,0 +1,9 @@
+html {
+    height: 100%;
```
Perhaps use `100vh` if you wish to limit to the full screen height.

FredrikAugust committed:
Use Story settings for controlling dashboard story

FredrikAugust commented on src/stories/DashboardItemSmall.stories.tsx:
```diff
@@ -0,0 +1,37 @@
+
+export default {
+    title: 'Dashboard/Dashboard Item Small',
+    component: DashboardItemSmall,
```
Use `Primary` as the component.

Also you don't need to use the `.bind` as far as I can see. 

Take a look at the rewritten `Dashboard.stories` file I pushed.

FredrikAugust reviewed (changes requested):
Pushing up some changes that we can take a look at tomorrow

FredrikAugust committed:
Helper to resolve grid positions with a reduce

FredrikAugust committed:
Comment out apple touch icon as it doesn’t exist

FredrikAugust committed:
Remove maximisation feature

Temporarily disabled. Left to frontend gang to figure out how to solve this without a lot of unncessary renders.

FredrikAugust commented:
Due to issues with performance, I've temporarily removed the min/max feature on the cards to simplify the currently semi-functional code.

There's still quite a lot of unnecessary renders being performed by the grid system, which you can see with the profiling features in the React browser toolkit, or by the `console.log` statements in your console.

FredrikAugust committed:
Remove magic margin/width from Pane component in Dashboard
